ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/point/DropPinTaskFragment.kt

```diff
@@ -52,7 +52,7 @@ class DropPinTaskFragment @Inject constructor() : AbstractTaskFragment<DropPinTa
     val args = Bundle()
     args.putString(TASK_ID_FRAGMENT_ARG_KEY, taskId)
     fragment.arguments = args
-    parentFragmentManager
+    childFragmentManager
```

Comment:
What does this do? Should this be in a separate PR in case it breaks something in prod?

Reply:
The relevance of parentFragmentManager vs childFragmentManager is dependent on what the child fragment needs access to. If the child fragment only needs to interact with the parent fragment, then `childFragmentManager` is needed. If the child fragment needs to interact with the activity or the parent fragment's siblings, then only `parentFragmentManager` is needed.

In our case, `childFragmentManager` is enough. This fixes the scope of the access. In addition to that, this also resolves the issue of unit tests. Earlier, the tests couldn't be added for the mapTaskFragments as the parent fragment's viewmodel couldn't be injected. This was because nav controller set for the parent fragment wasn't applied to the child fragment. However, this got auto fixed when using `childFragmentManager`. This was needed for adding tests for CaptureLocationTaskFragment. Since all map based tasks are same, I added the same change to drop pin and draw area tasks as well. Verified that the parent VM injection works fine even after this change.